pkg/diagnostics/systemd/locate_units.go

```diff
@@ -3,15 +3,35 @@ package systemd
 import (
 	"fmt"
 	"os/exec"
+	"runtime"
 	"strings"
 
 	"github.com/openshift/origin/pkg/diagnostics/log"
 	"github.com/openshift/origin/pkg/diagnostics/types"
 )
 
+// IsSystemd checks if this is a systemd host. We do this by checking that both systemctl and journald are
+// present and functional. When running in a container it is possible for systemctl to be
+// present but non-functional, or misleadingly functional due to sys mounts and a privileged container.
+func IsSystemd() bool {
+	if runtime.GOOS == "linux" {
+		systemctlErr := exec.Command("systemctl").Run()
+		journalctlErr := exec.Command("journalctl", "-n", "1").Run()
```

Comment:
I'll try to make a quick case for keeping them together if I may, for AnalyzeLogs and LocateUnits, it feels like they run/skip together. So as far as the caller is concerned, we just want to know if this is a systemd host or not. One method for answering this question gives us one spot to modify in the future if we find a way to more accurately check, and I could forsee that happening if we for example wanted to also check for the existence of at least one of our systemd units as part of answering that question. 

As an example, in all-in-one container today, systemctl will work if you run it correctly, but it's reporting against the host's systemd and in no way should we be considering that a systemd openshift host. Combining them helps clarify that this is not a functional systemd environment. 

Feel free to disagree in as much or as little detail as you like and I will split em. :)